front/lib/api/assistant/global_agents/configurations/research.ts

```diff
@@ -0,0 +1,440 @@
+</data_source_file_system_read_content>
+
+</data_source_file_system_api_instructions>
+
+<parallel_tool_calling_instructions>
+You can use parallel tool calling in order to speed-up your work, when applicable. Tasks that can be done in parallel should be done in parallel.
+</parallel_tool_calling_instructions>
+
+<output_guidelines>
+You must never output text outside of \`<thinking>\` tags between tool use. Only start writing in the main response body (in \`<response>\` tags) once you are done using tools and ready to write a final answer.
+</output_guidelines>
+`;
+
+  const researchAgent = {
+    id: -1,
+    sId: GLOBAL_AGENTS_SID.RESEARCH,
+    version: 0,
+    versionCreatedAt: null,
+    versionAuthorId: null,
+    name,
+    description,
+    instructions,
+    pictureUrl,
+    scope: "global" as const,
+    userFavorite: false,
+    model,
+    visualizationEnabled: true,
+    templateId: null,
+    requestedGroupIds: [],
+    tags: [],
+    canRead: true,
+    canEdit: false,
+  };
+
+  if (
+    (settings && settings.status === "disabled_by_admin") ||
+    !modelConfiguration
+  ) {
+    return {
+      ...researchAgent,
+      status: "disabled_by_admin",
+      actions: [
+        ..._getDefaultWebActionsForGlobalAgent({
+          agentId: GLOBAL_AGENTS_SID.RESEARCH,
+          webSearchBrowseMCPServerView,
+        }),
+      ],
+      maxStepsPerRun: 0,
+    };
+  }
+
+  // This only happens when we fetch the list version of the agent.
+  if (!preFetchedDataSources) {
+    return {
+      ...researchAgent,
+      status: "active",
+      actions: [],
+      maxStepsPerRun: MAX_STEPS_USE_PER_RUN_LIMIT,
+    };
+  }
+
+  const actions: MCPServerConfigurationType[] = [];
+
+  const dataSourceViews = preFetchedDataSources.dataSourceViews.filter(
+    (dsView) => dsView.dataSource.assistantDefaultSelected === true
+  );
+
+  // Only add the action if there are data sources and the search MCPServer is available.
+  if (dataSourceViews.length > 0 && searchMCPServerView) {
+    // We push one action with all data sources
+    actions.push({
+      id: -1,
+      sId: GLOBAL_AGENTS_SID.RESEARCH + "-datasource-action",
+      type: "mcp_server_configuration",
+      name: "search_all_data_sources",
+      description: "The user's entire workspace data sources",
+      mcpServerViewId: searchMCPServerView.sId,
+      internalMCPServerId: searchMCPServerView.internalMCPServerId,
+      dataSources: dataSourceViews.map((dsView) => ({
+        dataSourceViewId: dsView.sId,
+        workspaceId: preFetchedDataSources.workspaceId,
+        filter: { parents: null, tags: null },
+      })),
+      tables: null,
+      childAgentId: null,
+      reasoningModel: null,
+      additionalConfiguration: {},
+      timeFrame: null,
+      dustAppConfiguration: null,
+      jsonSchema: null,
+    });
+
+    // Add one action per managed data source to improve search results for queries like
```

Comment:
it should be a data source file system (only one) configured on all data sources at once